finops/resource_budget_test.go

```diff
@@ -0,0 +1,233 @@
+			},
+		},
+		Filter: &billing.BudgetConfigurationFilter{
+			Tags: []billing.BudgetConfigurationFilterTagClause{
+				{
+					Key: "Environment",
+					Value: &billing.BudgetConfigurationFilterClause{
+						Operator: billing.BudgetConfigurationFilterOperatorIn,
+						Values:   []string{"Testing"},
+					},
+				},
+			},
+			WorkspaceId: &billing.BudgetConfigurationFilterWorkspaceIdClause{
+				Operator: billing.BudgetConfigurationFilterOperatorIn,
+				Values: []int64{
+					1234567890098765,
+				},
+			},
+		},
+		BudgetConfigurationId: "budget_configuration_id",
+		DisplayName:           "budget_name",
+	}
+}
+
+func TestResourceBudgetCreate(t *testing.T) {
+	qa.ResourceFixture{
+		MockAccountClientFunc: func(a *mocks.MockAccountClient) {
+			api := a.GetMockbudgetsAPI().EXPECT()
+			api.Create(mock.Anything, billing.CreateBudgetConfigurationRequest{
+				Budget: billing.CreateBudgetConfigurationBudget{
+					AlertConfigurations: []billing.CreateBudgetConfigurationBudgetAlertConfigurations{
+						{
+							ActionConfigurations: []billing.CreateBudgetConfigurationBudgetActionConfigurations{
+								{
+									ActionType: getTestBudget().AlertConfigurations[0].ActionConfigurations[0].ActionType,
+									Target:     getTestBudget().AlertConfigurations[0].ActionConfigurations[0].Target,
+								},
+							},
+							QuantityThreshold: "840.84",
+							QuantityType:      getTestBudget().AlertConfigurations[0].QuantityType,
+							TimePeriod:        getTestBudget().AlertConfigurations[0].TimePeriod,
+							TriggerType:       getTestBudget().AlertConfigurations[0].TriggerType,
+						},
+					},
+					DisplayName: getTestBudget().DisplayName,
+					Filter:      getTestBudget().Filter,
+				},
+			}).Return(&billing.CreateBudgetConfigurationResponse{Budget: getTestBudget()}, nil)
+			api.GetByBudgetId(mock.Anything, "budget_configuration_id").Return(
+				&billing.GetBudgetConfigurationResponse{Budget: getTestBudget()}, nil,
+			)
+		},
+		Create:    true,
+		AccountID: "account_id",
+		HCL: `
+		display_name = "budget_name"
+	
+		alert_configurations {
+			time_period         = "MONTH"
+			trigger_type        = "CUMULATIVE_SPENDING_EXCEEDED" 
+			quantity_type       = "LIST_PRICE_DOLLARS_USD"
+			quantity_threshold  = "840.84"
+	
+			action_configurations {
+				action_type = "EMAIL_NOTIFICATION"
+				target      = "me@databricks.com"
+			}
+		}
+
+		filter {
+			tags {
+				key   = "Environment"
+				value {
+					operator = "IN"
+					values = ["Testing"]
+				}
+			}
+
+			workspace_id {
+				operator = "IN"
+				values   = [
+					1234567890098765
+				]                              
+			}
+		}
+		`,
+		Resource: ResourceBudget(),
+	}.ApplyAndExpectData(t, map[string]any{
+		"display_name":           "budget_name",
+		"id":                     "account_id|budget_configuration_id",
+		"alert_configurations.#": 1,
+		"filter.#":               1,
+	})
+}
+
+func TestResourceBudgetRead(t *testing.T) {
```

Comment:
Might be good to add some more tests to check the error pathways like we have for other resource eg: unpack(d) leading to error (i.e. id not supplied properly).